tox.ini

```diff
@@ -0,0 +1,111 @@
+[tox]
+minversion = 3.16.1
+envlist = lint,packaging,py{39,38,37,36}-{core,ansible29,devel}
+isolated_build = true
+requires =
+  setuptools >= 41.4.0
+  pip >= 19.3.0
+skip_missing_interpreters = True
+# `usedevelop = true` overrides `skip_install` instruction, it's unwanted
+usedevelop = false
+
+[testenv]
+description =
+  Run the tests under {basepython} and
+  devel: ansible devel branch
+  ansible29: ansible 2.9
+  core: ansible-base 2.10
+  py: ansible-core 2.11
+extras =
+  test
+deps =
+  ansible29: ansible>=2.9,<2.10
+  py: ansible-core>=2.11
+  devel: ansible-core @ git+https://github.com/ansible/ansible.git  # GPLv3+
+commands =
+  # safety measure to assure we do not accidentally run tests with broken dependencies
+  {envpython} -m pip check
+  # We add coverage options but not making them mandatory as we do not want to force
+  # pytest users to run coverage when they just want to run a single test with `pytest -k test`
+  {envpython} -m pytest \
+  --junitxml "{toxworkdir}/junit.{envname}.xml" \
+  {posargs:\
+    --cov ansiblecompat \
+    --cov "{envsitepackagesdir}/ansiblecompat" \
+    --cov-report term-missing:skip-covered \
+    --cov-report xml:.test-results/pytest/cov.xml \
+    --no-cov-on-fail}
+passenv =
+  CURL_CA_BUNDLE  # https proxies, https://github.com/tox-dev/tox/issues/1437
+  FORCE_COLOR
+  HOME
+  NO_COLOR
+  PYTEST_*  # allows developer to define their own preferences
+  PY_COLORS
+  REQUESTS_CA_BUNDLE  # https proxies
+  SSL_CERT_FILE  # https proxies
+  LANG
+  LC_ALL
+  LC_CTYPE
+# recreate = True
+setenv =
+  COVERAGE_FILE = {env:COVERAGE_FILE:{toxworkdir}/.coverage.{envname}}
+  PIP_DISABLE_PIP_VERSION_CHECK = 1
+  PIP_CONSTRAINT = {toxinidir}/constraints.txt
```

Comment:
It's a good idea to have this when building the dists so that sdist/wheel are reproducible (because this would force pinned PEP 517 build env)

Reply:
This is included by default for packaging (inhereted).